test/test_pythonkey.py

```diff
@@ -351,6 +356,9 @@ def get_grads(args):
             for arg in args:
                 arg.requires_grad = True
 
+            # Recompile since we're changing which inputs requires grads.
+            compiled_f = compiled_function(f, lambda x,_: x, lambda x,_: x)
```

Comment:
The purpose of these tests is to make sure the function we're tracing isn't overspecializing on inputs.

Can you replace this with
```
            def create_new_arg(x):
                if x.requires_grad:
                    return x.detach().uniform_(0, 1).requires_grad_()
                else:
                    return x.detach().uniform_(0, 1)
            args = pytree.tree_map(create_new_arg, args)
```

I think that also fixes some of the previous failures :P 

Alternately, I can just merge this first and then change it.